static/app/views/alerts/incidentRules/duplicate.tsx

```diff
@@ -0,0 +1,179 @@
+import {IncidentRule} from 'sentry/views/alerts/incidentRules/types';
+import {WizardRuleTemplate} from 'sentry/views/alerts/wizard/options';
+import AsyncView from 'sentry/views/asyncView';
+
+import RuleForm from './ruleForm';
+
+type RouteParams = {
+  orgId: string;
+  projectId?: string;
+  ruleId?: string;
+};
+
+type Props = {
+  eventView: EventView | undefined;
+  organization: Organization;
+  project: Project;
+  userTeamIds: string[];
+  sessionId?: string;
+  wizardTemplate?: WizardRuleTemplate;
+} & RouteComponentProps<RouteParams, {}>;
+
+type State = {
+  defaultRule: IncidentRule;
+  duplicateTargetRule?: IncidentRule;
+} & AsyncView['state'];
+
+/**
+ * Show metric rules form with values from an existing rule. Redirects to alerts list after creation.
+ */
+
+class IncidentRulesDuplicate extends AsyncView<Props, State> {
+  static duplicateMetricFields = [
+    'dataset',
+    'eventTypes',
+    'aggregate',
+    'query',
+    'timeWindow',
+    'thresholdPeriod',
+    'projects',
+    'environment',
+    'resolveThreshold',
+    'thresholdType',
+    'owner',
+    'name',
+    'projectId',
+    'comparisonDelta',
+  ];
+  static duplicateTriggerFields = ['alertThreshold', 'label'];
+  static duplicateActionFields = [
+    'type',
+    'targetType',
+    'targetIdentifier',
+    'inputChannelId',
+    'options',
+  ];
+
+  get isDuplicateRule() {
+    const {
+      location: {query},
+      organization,
+    } = this.props;
+    const hasDuplicateAlertRules = organization.features.includes('duplicate-alert-rule');
+    return hasDuplicateAlertRules && query.createFromDuplicate && query.duplicateRuleId;
+  }
+
+  getDefaultState() {
+    const {project, eventView, wizardTemplate, userTeamIds} = this.props;
+    const defaultRule = eventView
+      ? createRuleFromEventView(eventView)
+      : wizardTemplate
+      ? createRuleFromWizardTemplate(wizardTemplate)
+      : createDefaultRule();
+
+    const projectTeamIds = new Set(project.teams.map(({id}) => id));
+    const defaultOwnerId = userTeamIds.find(id => projectTeamIds.has(id)) ?? null;
+    const owner = defaultOwnerId && `team:${defaultOwnerId}`;
+
+    return {
+      ...super.getDefaultState(),
+      defaultRule: {
+        ...defaultRule,
+        owner,
+        projects: [project.slug],
+      },
+    };
+  }
+
+  getEndpoints(): ReturnType<AsyncView['getEndpoints']> {
+    const {
+      params: {orgId},
+      location: {query},
+    } = this.props;
+
+    if (this.isDuplicateRule) {
+      return [
+        [
+          'duplicateTargetRule',
+          `/organizations/${orgId}/alert-rules/${query.duplicateRuleId}/`,
+        ],
+      ];
+    }
+
+    return [];
+  }
+
+  handleSubmitSuccess = (data: any) => {
+    const {
+      router,
+      project,
+      params: {orgId},
+    } = this.props;
+    const alertRuleId: string | undefined = data
+      ? (data.id as string | undefined)
+      : undefined;
+
+    metric.endTransaction({name: 'saveAlertRule'});
```

Comment:
lets remove the metric recording here, so that we don't have data coming from different places

Reply:
👍